echoregions/regions2d/regions2d.py

```diff
@@ -73,42 +73,140 @@ def to_json(self, save_path: str = None) -> None:
         """
 
     def select_region(
-        self, region: Union[float, str, list, Series, DataFrame] = None, copy=False
+        self,
+        region_id: Union[float, int, str, List[Union[float, int, str]]] = None,
+        time_range: List[Timestamp] = None,
+        depth_range: List[Union[float, int]] = None,
+        copy=False,
     ) -> DataFrame:
-        """Ensure that region is a DataFrame.
+        """Selects a subset of this region object's dataframe.
 
         Parameters
         ----------
-        region : float, str, list, Series, DataFrame, ``None``
+        region_id : float, int, str, list, ``None``
             A region id provided as a number, string, list of these,
             or a DataFrame/Series containing the region_id column name.
+        time_range: List of 2 Pandas Timestamps.
+            Datetime range for expected output of subselected DataFrame. 1st
+            index value must be later than 0th index value.
+        depth_range: List of 2 floats.
+            Depth range for expected output of subselected DataFrame. 1st
+            index value must be larger than 0th index value.
         copy : bool
             Return a copy of the `data` DataFrame
         Returns
         -------
         DataFrame
             A DataFrame subselected from Regions2D.data.
-            There is a row for each region id provided by the region parameter.
+            There is a row for each region id provided by the region_id parameter,
+            and each row has time and depth within or on the boundaries passed
+            in by the time_range and depth_range values.
```

Comment:
Just some formatting stuff
```suggestion
            There is a row for each region id provided by the ``region_id`` parameter,
            and each row has time and depth within or on the boundaries passed
            in by the ``time_range`` and ``depth_range`` values.
```